src/compiler/checker.ts

```diff
@@ -3006,637 +3024,171 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
-                        //        resolve it as an alias. This will cause the checker to consider the export specifier
-                        //        a circular alias reference when it might not be.
-                        //     2. We check === SymbolFlags.Alias in order to check that the symbol is *purely*
-                        //        an alias. If we used &, we'd be throwing out symbols that have non alias aspects,
-                        //        which is not the desired behavior.
-                        const moduleExport = moduleExports.get(name);
-                        if (
-                            moduleExport &&
-                            moduleExport.flags === SymbolFlags.Alias &&
-                            (getDeclarationOfKind(moduleExport, SyntaxKind.ExportSpecifier) || getDeclarationOfKind(moduleExport, SyntaxKind.NamespaceExport))
-                        ) {
-                            break;
-                        }
-                    }
-
-                    // ES6 exports are also visible locally (except for 'default'), but commonjs exports are not (except typedefs)
-                    if (name !== InternalSymbolName.Default && (result = lookup(moduleExports, name, meaning & SymbolFlags.ModuleMember))) {
-                        if (isSourceFile(location) && location.commonJsModuleIndicator && !result.declarations?.some(isJSDocTypeAlias)) {
-                            result = undefined;
-                        }
-                        else {
-                            break loop;
-                        }
-                    }
-                    break;
-                case SyntaxKind.EnumDeclaration:
-                    if (result = lookup(getSymbolOfDeclaration(location as EnumDeclaration)?.exports || emptySymbols, name, meaning & SymbolFlags.EnumMember)) {
-                        if (nameNotFoundMessage && getIsolatedModules(compilerOptions) && !(location.flags & NodeFlags.Ambient) && getSourceFileOfNode(location) !== getSourceFileOfNode(result.valueDeclaration)) {
-                            error(
-                                errorLocation,
-                                Diagnostics.Cannot_access_0_from_another_file_without_qualification_when_1_is_enabled_Use_2_instead,
-                                unescapeLeadingUnderscores(name),
-                                isolatedModulesLikeFlagName,
-                                `${unescapeLeadingUnderscores(getSymbolOfNode(location)!.escapedName)}.${unescapeLeadingUnderscores(name)}`,
+        nameNotFoundMessage: DiagnosticMessage,
+    ) {
+        const name = typeof nameArg === "string" ? nameArg : (nameArg as Identifier).escapedText;
+        addLazyDiagnostic(() => {
+            if (
+                !errorLocation ||
+                errorLocation.parent.kind !== SyntaxKind.JSDocLink &&
+                    !checkAndReportErrorForMissingPrefix(errorLocation, name, nameArg) && // TODO: GH#18217

```

Comment:
```suggestion
                    !checkAndReportErrorForMissingPrefix(errorLocation, name, nameArg) &&
```

(If we remove a `!`, the TODO can go away.)